src/zenml/integrations/airflow/orchestrators/airflow_orchestrator.py

```diff
@@ -447,224 +439,8 @@ def _translate_schedule(
-
-        fileio.rmtree(self.airflow_home)
-        logger.info("Airflow spun down.")
-
-    def _set_server_env(self) -> None:
-        """Sets environment variables for the local Airflow server process."""
-        os.environ["AIRFLOW__CORE__DAGS_FOLDER"] = self.dags_directory
-        os.environ["AIRFLOW__CORE__LOAD_EXAMPLES"] = "false"
-        # check the DAG folder every 10 seconds for new files
-        os.environ["AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL"] = "10"
-
-        if platform.system() == "Darwin":
-            # Prevent crashes during forking on MacOS
-            # https://github.com/apache/airflow/issues/28487
-            os.environ["no_proxy"] = "*"
-
-    @staticmethod
-    def _check_local_server_requirements() -> None:
-        """Checks that all packages for a local Airflow server are installed.
-
-        When running a local Airflow server, we require the
-        `apache-airflow-providers-docker` to run steps locally in Docker
-        containers in addition to the basic integration requirements.
-
-        Raises:
-            RuntimeError: If the `apache-airflow-providers-docker` is not
-                installed in the active Python environment.
-        """
-        try:
-            from airflow.providers.docker.operators.docker import (  # noqa
-                DockerOperator,
-            )
-        except ImportError:
-            raise RuntimeError(
-                "Unable to import Airflow `DockerOperator` in the active "
-                "Python environment. Spinning up a local Airflow server to "
-                "run ZenML pipelines requires the `DockerOperator` to be "
-                "available. Please run "
-                "`pip install 'apache-airflow-providers-docker<3.8.0' "
-                "'apache-airflow~=2.4.0'` to install it and try again."
-            )
-
-    def _log_webserver_credentials(self) -> None:
-        """Logs URL and credentials to log in to the airflow webserver.
-
-        Raises:
-            FileNotFoundError: If the password file does not exist.
-        """
-        if fileio.exists(self.password_file):
-            with open(self.password_file) as file:
-                password = file.read().strip()
-        else:
-            raise FileNotFoundError(
-                f"Can't find password file '{self.password_file}'"
-            )
-        logger.info(
-            "To inspect your DAGs, login to `http://localhost:8080` "
-            "with username: `admin` password: `%s`",
-            password,
-        )
-
-    def get_pipeline_run_metadata(
-        self, run_id: UUID
-    ) -> Dict[str, "MetadataType"]:
-        """Get general component-specific metadata for a pipeline run.
-
-        Args:
-            run_id: The ID of the pipeline run.
-
-        Returns:
-            A dictionary of metadata.
-        """
-        if self.config.local:
-            return {
-                METADATA_ORCHESTRATOR_URL: Uri("http://localhost:8080"),
```

Comment:
In case of a local airflow, can we somehow figure out this port and still include the URL?

Reply:
Also putting this here as I can't comment on the line directly: In `AirflowOrchestrator._write_dag` there's a warning log message that needs to be updated. Instead of warning users when they use a custom DAG directory, we should probably warn them when they don't use one. Even this warning might be wrong though, in case they configure their local Airflow deployment DAG directory in the correct way.

Reply:
Let's address this during the fix/test session.

Reply:
@schustmi Can we resolve this now?